tests/reconfiguration.py

```diff
@@ -286,7 +286,7 @@ def test_add_node_endorsements_endpoints(network, args):
                 new_node,
                 args.package,
                 args_copy,
-                timeout=15,
+                timeout=16,
```

Comment:
I claim that this test was somewhere between extremely fragile and outright wrong already, this PR messes with the timing just enough that we get into a consistent failure, so this tiny bump of a random timeout is a reasonable tradeoff to unblock this PR.

The _actual_ fix is to change the `snp_endorsements_servers` logic to honour the requested per-server retries, and scale this timeout accordingly. Currently the last run here passes 2 servers that we expect to fail, requests 4 retries for each (from infra code elsewhere), and the actual result is exactly 4 waits of a ~3s timeout before we throw a fatal error. And after the error the node pauses for a second anyway!